client/src/controllers/DrilldownController.test.js

```diff
@@ -0,0 +1,214 @@
+import { Application } from '@hotwired/stimulus';
+import { DrilldownController } from './DrilldownController';
+import { DropdownController } from './DropdownController';
+import { ActionController } from './ActionController';
+
+jest.useFakeTimers();
+
+describe('DrilldownController', () => {
+  const eventNames = [
+    'w-drilldown:opened',
+    'w-drilldown:closed',
+    'w-drilldown:ready',
+    'w-drilldown:toggled',
+  ];
+
+  const events = {};
+  let application;
+  let errors = [];
+  let controller;
+  let actionElement;
+
+  const setup = async (
+    html = `<section>
```

Comment:
If possible, are you able to clean up the HTML used in these test snippets.

1. Try to focus on the minimum DOM needed (e.g. remove most classes, unless they are used for testing or to help differentiate elements).
2. Change all nested spacing to just two spaces, it makes the HTML much more readable
3. try to align the opening/closing tags so that it's easier to spot any DOM issues
4. A reminder that `className` will not work here, it's `class` when working with HTML

Reply:
Okay, I will clean up and format the HTML as you mentioned. Regarding your fourth point, my bad,I had previously written `class` instead of` className`. When I ran `npm run lint` to check for issues, it suggested using `className` instead of `class`, but this only applies to stories.js where I used React. I'll make the change.